src/libcore/result.rs

```diff
@@ -539,9 +538,12 @@ impl<T, E> Result<T, E> {
     /// assert_eq!(x.map_or_else(|e| k * 2, |v| v.len()), 42);
     /// ```
     #[inline]
-    #[unstable(feature = "result_map_or_else", issue = "53268")]
-    pub fn map_or_else<U, M: FnOnce(T) -> U, F: FnOnce(E) -> U>(self, fallback: F, map: M) -> U {
-        self.map(map).unwrap_or_else(fallback)
```

Comment:
What was wrong with the previous implementation?

Reply:
The new implementation is consistent with `Option::map_or_else`: https://doc.rust-lang.org/nightly/src/core/option.rs.html#491-496

So we are following Principle of Least Surprise.

Reply:
Even less Surprising would be to not change an implementation in a stabilization PR.

Also, IMO the original implementation was better because it more clearly shows the relation between `map_or_else`, `map`, and `unwrap_or_else`. The whole point of these methods is that we don't have to write` match`es everywhere.

Reply:
If you like it, we could send a PR later to fix it in both Option and Result.

Reply:
@timvermeulen That might be true, but following the paradigm of keeping things simple and consistent is also needed. The original idea was to create map_or_else and unwrap_or_else the same as on Option, and then things get switched around. That is just absurd, and makes it harder to deal with because you to constantly ask yourself, "which way arround was this command?!" - and honestly i dont have time for that in a progeramming language.

Reply:
Is there a reason why Option::map_or_else has its parameters in the order it does? It looks pretty odd at first glance.

Reply:
@tomwhoiscontrary I think that question has been asked before, or a derivative thereof. The conclusion was that `Option`s `map_or_else` is already in stable and would therefore not be changed, as that would be breaking much more.
Part of the discussion is in here #53268

Reply:
@KatsuoRyuu My question is the one i asked - why is Option::map_or_else the way it is. I am not suggesting changing it. I want to understand why it is the way it is.

Reply:
@tomwhoiscontrary sorry for the misunderstanding that, will see if I can find the documentation for that for you, I think I saw it somewhere ;)

Reply:
@tomwhoiscontrary this is the thread I remembered seeing, #1025. So the reason for the initially is to follow the convention of map_or, so all parameters are allways prepended.

Reply:
The argument for having closures at the end is ergonomic.  Closures are typically longer syntactically than the other arguments and are more likely to wrap.  You see this preference for putting closures at the end of argument lists in Lisp and other languages with closures.  It's compelling ergonomics.  Rust decided to follow this convention long ago, and it would be an error to ever introduce inconsistencies with it.

Reply:
Thanks! The explanation makes sense. If we had keyword arguments, this problem would disappear!